### PR TITLE
Jones/more awards

### PR DESCRIPTION
**Pull request by taylor-jones**
1) Extends award page to integrate award table and award form.
2) Uses cookies to make award list request and only show awards created by the cookieUser (unless the cookieUser is admin, in which case it shows all awards).
3) Restricts an admin user from creating or updating an award.
4) Restricts any user from updating an award that was not created by that user.

### CODE REVIEW

**Comment by taylor-jones** [comment]
PDF Awards are now emailed automatically to the employee's email address whenever a new Award record gets created. Right now, the Award Certificate has no styling and is not visually appealing at all, but it does include all the award information and the user's signature.

Edit: The PDF generation also doesn't use LateX right now.

File: erp-client/src/app/app.module.ts [hunk 1]
```diff
@@ -48,7 +50,7 @@ import { SelfComponent } from './components/self/self.component';
     // NoopAnimationsModule <--- causes animations to not work if imported
   ],
   exports: [],
-	providers: [CookieService],
+	providers: [ CookieService ],
```
Comment (taylor-jones) [inline erp-client/src/app/app.module.ts:53]: Apologies for small styling changes like this. I think it got auto-formatted.

File: erp-client/src/app/components/login/login.component.ts [hunk 2]
```diff
@@ -17,9 +17,11 @@ export class LoginComponent implements OnInit {
...
   showSpinner = false;
 
-  constructor(private authService: AuthenticationService,
-              private router: Router,
-              private cookieService: CookieService){ }
+  constructor(
+    private authService: AuthenticationService,
+    private router: Router,
+    private cookieService: CookieService
+  ) {}
```
Comment (taylor-jones) [inline erp-client/src/app/components/login/login.component.ts:24]: Same here w/ the formatting. Not trying to change anyone else's format.

File: tempCodeRunnerFile.javascript [hunk 3]
```diff
@@ -0,0 +1,2 @@
+const time = Date.parse("20:00:00")
```
Comment (taylor-jones) [inline tempCodeRunnerFile.javascript]: Whoops. This file shouldn't have been committed.

File: erp-server/src/main/java/com/ttt/erp/service/AwardService.java [hunk 4]
```diff
@@ -44,17 +76,20 @@
...
             Award toDelete = this.repository.findById(awardId);
             logDelete(userAccountId, toDelete.getClass().getSimpleName(), awardId);
             this.repository.delete(toDelete);
-            return new ResponseEntity<>("Success", HttpStatus.OK);
+            return ResponseEntity.ok().build();
         } catch (Exception e) {
-            System.err.println("Error on Award update");
-            e.printStackTrace();
-            return new ResponseEntity<>("Failed", HttpStatus.BAD_REQUEST);
+            return ResponseEntity.notFound().build();
```
Comment (tmrietz) [inline erp-server/src/main/java/com/ttt/erp/service/AwardService.java:92]: I like this pattern much better, nice.

File: erp-server/src/main/java/com/ttt/erp/service/AuthenticationSuccessHandler.java [hunk 5]
```diff
@@ -29,11 +28,14 @@ public void onAuthenticationSuccess(HttpServletRequest request, HttpServletRespo
         UserAccount userAccount = this.userAccountService.getUserByUsername(authentication.getName());
         if (userAccount != null) {
             this.userManager.set(userAccount);
-            Cookie userCookie = new Cookie("user", userAccount.getUsername().toString());
+            Cookie userCookie = new Cookie("user", userAccount.getUsername());
+            Cookie userIdCookie = new Cookie("userId", userAccount.getId().toString());
```
Comment (tmrietz) [inline erp-server/src/main/java/com/ttt/erp/service/AuthenticationSuccessHandler.java:32]: Nice 👍

File: erp-client/src/app/services/snackbar/snackbar.service.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+   * Shows a snackbar with an error message
+   * @param {string} message - the message to display on the smackbar
+   */
+  public showError(message, action = 'Okay', options = {
+    duration: 3000,
+    panelClass: [ 'snackbar-success' ]
+  }) {
+    this._snackbar.open(message, action, options);
+  }
+}
```
Comment (tmrietz) [inline erp-client/src/app/services/snackbar/snackbar.service.ts:34]: I like that you pulled these into their own service. Nice job.

File: erp-server/src/main/java/com/ttt/erp/service/AwardPdfService.java [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+
+    public ResponseEntity<Long> sendAwardPdf(Long id) throws FileNotFoundException {
+        try {
+            // gather the components needed to build the file name
+            Award award = repository.findById(id);
+            Employee employee = award.getEmployee();
+
+            String filename = employee.getFullName() + " - Award.pdf";
+            String subjectLine = "Congratulations on award from TTT";
```
Comment (tmrietz) [inline erp-server/src/main/java/com/ttt/erp/service/AwardPdfService.java]: Only comment here is that may be good to say "...from \<user who created award\>" in the subject and message. That's a total nitpick though, this all looks really good.

**Review by tmrietz — APPROVED** [review]
Really good work here! Looks good.

File: erp-server/src/main/java/com/ttt/erp/service/AwardPdfService.java (hunk 7 above)
Comment (taylor-jones) [reply erp-server/src/main/java/com/ttt/erp/service/AwardPdfService.java]: Good idea. I'll make the change.

**Pull request merged**